sentry-ruby/lib/sentry-ruby.rb

```diff
@@ -248,35 +251,37 @@ def init(&block)
     #
     # @return [void]
     def close
-      if @session_flusher
-        @session_flusher.flush
-        @session_flusher.kill
-        @session_flusher = nil
-      end
+      @mutex.synchronize do
+        if @session_flusher
+          @session_flusher.flush
+          @session_flusher.kill
+          @session_flusher = nil
+        end
 
-      if @backpressure_monitor
-        @backpressure_monitor.kill
-        @backpressure_monitor = nil
-      end
+        if @backpressure_monitor
+          @backpressure_monitor.kill
+          @backpressure_monitor = nil
+        end
 
-      if @metrics_aggregator
-        @metrics_aggregator.flush(force: true)
-        @metrics_aggregator.kill
-        @metrics_aggregator = nil
-      end
+        if @metrics_aggregator
+          @metrics_aggregator.flush(force: true)
+          @metrics_aggregator.kill
+          @metrics_aggregator = nil
+        end
 
-      if client = get_current_client
-        client.flush
+        if client = get_current_client
+          client.flush
 
-        if client.configuration.include_local_variables
-          exception_locals_tp.disable
+          if client.configuration.include_local_variables
+            exception_locals_tp.disable
+          end
         end
-      end
 
-      @background_worker.shutdown
+        @background_worker.shutdown
 
-      @main_hub = nil
-      Thread.current.thread_variable_set(THREAD_LOCAL, nil)
+        @main_hub = nil
```

Comment:
let's just wrap lines 282 and 283 in the mutex, the other stuff is fine.
And I agree that `get_main_hub` should also be within a mutex sync